python/ray/data/preprocessor.py

```diff
@@ -326,3 +327,14 @@ def preferred_batch_format(cls) -> BatchFormat:
         path is the most optimal.
         """
         return BatchFormat.PANDAS
+
+    @DeveloperAPI
+    def pickle(self) -> str:
+        """Return this preprocessor pickled as a string."""
+        return pickle.dumps(self, 0).decode("ascii")
```

Comment:
I would leave this as the default protocol (i.e. pickle.DEFAULT_PROTOCOL) since protocol 0 is really space inefficient (some kind of text protocol I believe). I don't think there is any advantage of using 0 here but lmk if you think there is :)

Reply:
I believe we have to use this protocol if we want to ensure it's ASCII decodable. Alternatively we could base64 the string, though this way seems a bit less opaque.

Reply:
Ah lol, but it also doesn't work --
```python
pickle.dumps('á', 0).decode("ascii")
```
gives an error (this is basically a Python bug https://bugs.python.org/issue2980). Why don't we use the default protocol and base64, then we are safe (and not too verbose) :)

Reply:
Ah one thing I learned from @justinvyu is to use these utils:

```
from ray._private.utils import binary_to_hex, hex_to_binary
```

https://github.com/ray-project/ray/blob/41e34a15cea055f3f4bbacae0b7bc46dd5b211b4/python/ray/train/_internal/framework_checkpoint.py#L28-L45

Would that work?

Reply:
Fixed.

Reply:
@matthewdeng that would work, but seems the `base64` module also works.